### PR TITLE
Add JavaScript `HTMLProcessor` class

### PR DESCRIPTION
**Pull request by kojiishi**
This class provides extended capabilities when applying to DOM,
but with larger code size. Please see the JSDoc for more details.

### CODE REVIEW

File: javascript/README.md [hunk 1]
```diff
@@ -63,6 +63,21 @@ console.log(ele.outerHTML);
 // <p class="budou-this" style="word-break: keep-all; overflow-wrap: break-word;">今日は<b><wbr>とても<wbr>天気</b>です。</p>
 ```
 
+There is another way to apply the process to an HTML element.
+
+```javascript
```
Comment (tushuhei) [inline javascript/README.md:68]: Let's add the line `import { Applier } from 'budoux';` to show how users can import the Applier class.

File: javascript/src/applier.ts [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+  hasText(): boolean {
+    return this.textNodes.length > 0;
+  }
+}
+
+/**
+ * Applies the BudouX to the given DOM.
+ *
+ * This class has following advantages over
+ * {@link Applier.applyToElement}.
```
Comment (tushuhei) [inline javascript/src/applier.ts:143]: Probably this should be `parser.applyElement` instead?

File: javascript/src/applier.ts [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+ * * It recognizes paragraphs and applies the BudouX for each
+ *   paragraph separately.
+ * * It can customize how to insert break opportunities.
+ *   See {@link separator} for more details.
+ * * It is generally faster and more memory efficient, but the
+ *   code size is larger.
+ */
+export class Applier {
```
Comment (tushuhei) [inline javascript/src/applier.ts:151]: I thought we could rename this class more specifically to HTML to secure a space for other possible platforms.
How about `HTMLProcessor` for example?

File: javascript/src/applier.ts [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+   * that defines the default styles as a class.
+   *
+   * When falsy, an inline style is set instead.
+   */
+  className?: string;
+
+  /**
+   * @param parser A BudouX {@link Parser} to compute semantic line breaks.
+   */
+  constructor(parser: Parser) {
```
Comment (tushuhei) [inline javascript/src/applier.ts:179]: Can we add an `options` arg to make those properties (e.g. `separator`) configurable when instantiating one?

**Review by tushuhei — COMMENTED** [review]
Thank you for your PR!

File: javascript/README.md (hunk 1 above)
Comment (kojiishi) [reply javascript/README.md:68]: Done, thanks for the good suggestion.

File: javascript/src/applier.ts (hunk 2 above)
Comment (kojiishi) [reply javascript/src/applier.ts]: Fixed, thanks!

File: javascript/src/applier.ts (hunk 3 above)
Comment (kojiishi) [reply javascript/src/applier.ts]: `HTMLProcessor` sounds better than `Applier`, thanks!

File: javascript/src/applier.ts (hunk 4 above)
Comment (kojiishi) [reply javascript/src/applier.ts]: Good idea, thanks, done.

**Review by tushuhei — APPROVED** [review]
Thank you for the changes!

**Pull request merged**